src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,291 @@
+                address_octets = address_data['address'].split('.')
+                if address_octets[0] == gateway_octets[0] \
+                        and address_octets[1] == gateway_octets[1] \
+                        and address_octets[2] == gateway_octets[2]:
+
+                    ip_address = address_data['address']
+
+            if ip_address is None:
+                self.logger.warning(
+                    'No IP addresses for connection %s associated with a gateway.',
+                    connection_id)
+
+        else:
+            self.logger.warning(
+                'Attempted to get IP address for connection %s, which is not connected.',
+                connection_id)
+
+        return ip_address
+
+    def connection_is_activated(self, connection_id):
+        """Check whether the network with the given connection id is ready.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+
+        connection_is_activated = False
+
+        connection_state = self._get_connection_state(connection_id)
+
+        if connection_state is NM_CONNECTION_ACTIVATED:
+            connection_is_activated = True
+
+        return connection_is_activated
+
+    def _build_connection_id_table(self):
+        """Assemble a helpful dictionary of connection objects, indexed by the connection's
+        id in NetworkManager.
+        """
+        connection_id_to_connection_dict = {}
+
+        all_connections = NetworkManager.Settings.ListConnections()
+        self._run_proxy_call(all_connections)
+
+        for connection in all_connections:
+            connection_id = connection.GetSettings()['connection']['id']
+            connection_id_to_connection_dict[connection_id] = connection
+
+        return connection_id_to_connection_dict
+
+    def _create_device_objects(self, wired_interface_name, wifi_interface_name):
+        """Store references to the device objects that are used frequently.
+
+        wired_interface_name: The name of the wired network interface, e.g. eth0, enp0s1.
+        wifi_interface_name: The name of the wifi network interface, e.g. wlan0, wlp0s1.
+        """
+        wired_device = None
+        wireless_device = None
+
+        for device in NetworkManager.NetworkManager.GetDevices():
+            if device.Interface == wired_interface_name:
+                wired_device = device
+
+            if device.Interface == wifi_interface_name:
+                wireless_device = device
+
+        if wired_device is None:
+            raise DeviceNotFoundException('Configured wired device %s was not found.' %
+                                          wired_interface_name)
+
+        if wireless_device is None:
+            raise DeviceNotFoundException('Configured wireless device %s was not found.' %
+                                          wifi_interface_name)
+
+        return (wired_device, wireless_device)
+
+    def _get_device_for_connection(self, connection):
```

Comment:
This should probably just take a connection_id and not a full connection. We just use it to get the connection_id away and everywhere this method is called, the connection_id is already available.

Reply:
Oops, this was left over from a previous version of _get_device_for_connection.